Clean up Dev hub's dashboard image and install mystmd,voila and voici packages

balajialg opened the pull request:
Ref:
https://anaconda.org/conda-forge/voila
https://anaconda.org/conda-forge/voila
https://anaconda.org/conda-forge/mystmd


~Image was building for a long time locally (tried multiple iterations locally - it kept building. So, wanted to check whether it built via CI)~

~Okay, debugging locally~

Commented out unnecessary packages for the dashboard image and added  above packages which are needed to build dashboard. When I tested this image locally, it built successfully.

<img width="563" alt="Screenshot 2024-07-10 at 4 54 16 PM" src="https://github.com/berkeley-dsep-infra/datahub/assets/2306166/a537f07a-d607-4190-8592-dce30aaac894">

balajialg committed:
Add voila and voici package in dashboard image

balajialg committed:
Install voici, mystmd and comment out unnecessary packages for dashboard image

ryanlovett commented on deployments/dev/images/secondary/environment.yml:
```diff
@@ -7,69 +7,69 @@ channels:
 dependencies:
 - python==3.11.*
 - git==2.39.1
+- jupyterhub==5.0.0
```
This does not match the version of jupyterhub in `images/hub/Dockerfile`. Our image is using 3.3.7 and according to https://hub.jupyter.org/helm-chart/ that corresponds with jupyterhub 4.1.5. That is also what all of our other images use. (the ones under `deployments/` at least)

balajialg replied:
Thanks Ryan, I will fix that now.

ryanlovett commented on deployments/dev/images/secondary/environment.yml:
```diff
@@ -7,69 +7,69 @@ channels:
 - notebook==7.0.7
 - folium==0.14.0
-- h5netcdf==1.0.2
+  #- h5netcdf==1.0.2
```
Feel free to delete old entries rather than comment them out, unless you're not certain and need a recent reminder. We can always look back into git history if needed. Sometimes I leave larger commented stanzas in as kind of a template.

balajialg replied:
Thanks Ryan! It makes sense. I have removed the following entries which are commented out from the yml file.

  #- h5netcdf==1.0.2
  #- mdit-py-plugins==0.4.0
  #- jupysql==0.8.0
  #- scikit-learn==1.2.2
  #- scipy==1.10.1
# Spring 2024 table demos
#- lxml==5.1.0
# Spring 2024 data 100
#- pytorch==2.1.2
#- cpuonly==2.0
#- transformers==4.37.1
  #- statsmodels==0.14.0
  #- tensorflow-cpu==2.12.1
  #- sqlalchemy==2.0.16
  #- mlxtend==0.23.0
  #- lifelines==0.27.8
  #- pytest-notebook==0.10.0
  #- gh-scoped-creds==4.1
  #- ydata-profiling==4.8.3
  #- otter-grader==5.5.0
  #- python-duckdb==1.0.0
  #- pip
  #- pip:
  # - -r infra-requirements.txt
  # disable until fixed (probably this:  https://github.com/jupyterlab/jupyter-collaboration/issues/162)
  # - jupyter_collaboration==1.0.1
  #- jupyterhub==4.1.5
  # - nbconvert[webpdf]
  #  - pyppeteer==2.0.0
  #- pytest-notebook==0.8.1
  #- gh-scoped-creds==4.1
  # - git+https://github.com/shaneknapp/python-popularity-contest.git@add-error-handling
  #- ydata-profiling==4.6.4
  #- otter-grader==5.4.0
  #- duckdb==0.10.1
  #  - duckdb_engine==0.11.2

ryanlovett commented on deployments/dev/images/secondary/environment.yml:
```diff
@@ -7,69 +7,69 @@ channels:
 - matplotlib==3.7.1
-- mdit-py-plugins==0.4.0
+# Install myst markdown for generating dashboards
+- mystmd==1.2.5
```
I think you can bump mystmd to 1.2.9.

balajialg replied:
Will do it now!

ryanlovett commented on deployments/dev/images/secondary/environment.yml:
```diff
@@ -7,69 +7,69 @@ channels:
-- quarto==1.4.550
-- pip
-- pip:
+- voici==0.6.1
```
voici is available as a conda package so should not be under pip.

balajialg replied:
Actually, voici is part of conda block. I am not sure why the Github snapshot shows it is under pip

ryanlovett reviewed (changes requested):
Thanks for adding the dashboard packages! I've left some notes. The most important one is the jupyterhub version.

balajialg committed:
Addressing review comments

balajialg commented:
@ryanlovett Thanks for the detailed feedback. I just created a PR addressing all your comments. The image built successfully in my local docker setup.

balajialg commented:
I am merging the changes to staging inorder to test it
